shell/common/platform_message_handler.h

```diff
@@ -42,6 +42,19 @@ class PlatformMessageHandler {
   /// HandlePlatformMessage where there is no return value.
   /// This method should be thread-safe and able to be invoked on any thread.
   virtual void InvokePlatformMessageEmptyResponseCallback(int response_id) = 0;
+
+  virtual bool AddPlatformPortCallback(int64_t port,
```

Comment:
In what cases can this fail?

Same nits on naming, you aren't adding a callback you are associating a port with a channel name. Please fix here and everywhere.